### ISSUE TITLE
Problem with Method definitions and "space-before-function-paren"

### ISSUE THREAD

**Issue by egoist** [eslint/eslint#4619]
When I set the rule `"space-before-function-paren": [2, "always"]`, the following works well (it shows error: `Missing space before function parentheses`):

```javascript
const foo = {
  bar() {
  }
};
```

but this (which there is a space before function paren):

```javascript
const foo = {
  bar () {
  }
};
```

throws an unexpected error:

```bash
C:\Users\EGOIST\dev\evangelion\node_modules\eslint\lib\rules\space-before-keywords.js:187
            if (right.type !== "Keyword") {
                     ^

TypeError: Cannot read property 'type' of undefined
    at EventEmitter.FunctionExpression (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\lib\rules\space-before-keywords.js:187:22)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.enterNode (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\lib\util\node-event-generator.js:42:22)
    at CommentEventGenerator.enterNode (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\lib\util\comment-event-generator.js:98:23)
    at Controller.controller.traverse.enter (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\lib\eslint.js:767:36)
    at Controller.__execute (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\node_modules\estraverse\estraverse.js:397:31)
    at Controller.traverse (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\node_modules\estraverse\estraverse.js:495:28)
    at EventEmitter.module.exports.api.verify (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\lib\eslint.js:764:24)
    at processText (C:\Users\EGOIST\dev\evangelion\node_modules\eslint\lib\cli-engine.js:230:27)
```

My config:
```json
"eslint": "^1.10.3",
"babel-eslint": "^5.0.0-beta4",
"eslint-config-airbnb": "^2.0.0"
```

**Comment by eslintbot** [comment]
@egoist Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Can you try it without `babel-eslint` and see if you have the same issue with default parser?

**Comment by egoist** [comment]
@ilyavolodin thx for the reply, using default parser solves this issue, definitely a` babel-eslint` issue then.
